Assets/Scripts/UI Scripts/HintTextManager.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class HintTextManager : MonoBehaviour
+{
+    [SerializeField]
+    private string movementHint = "Movement hint text will be here.";
+    [SerializeField]
+    private float movementHintDelay = 1;
+
+    [SerializeField]
+    private string shootHint = "Shooting hint text will be here.";
+    [SerializeField]
+    private float shootHintDelay = 10;
+
+    [SerializeField]
+    private string jumpHint = "Jumping hint text will be here.";
+    [SerializeField]
+    private float jumpHintDelay = 5;
+
+    [SerializeField]
+    private string lockHint = "Movement locking hint text will be here.";
+    [SerializeField]
+    private float lockHintDelay = 30;
+
+    [SerializeField]
+    private string aimHint = "Aiming hint text will be here.";
+    [SerializeField]
+    private float aimHintDelay = 20;
+
+    [SerializeField]
+    private float hintTime = 5;
+
+    [SerializeField]
+    private GameObject hintTextObject;
+
+    private TMP_Text hintTextComponent;
+
+    private GameObject player;
+    private PlayerController playerController;
+
+    private bool playerHasSeenMoveHint = false;
+    private bool playerHasSeenShootHint = false;
+    private bool playerHasSeenJumpHint = false;
+    private bool playerHasSeenLockHint = false;
+    private bool playerHasSeenAimHint = false;
+    private enum HintType
```

Comment:
I wish that I had some suggestions as to how to do this better, but sadly I don't. However, I will say that your code in general is much less messy than mine, if that gives you any encouragement haha. Only thing I would say is don't forget to add summaries and tool tips before the final release!

Reply:
You could create a hint  text class with a property HasBeenSeen